generator/src/main/java/com/graphicsfuzz/generator/mutateapi/ExpressionGenerator.java

```diff
@@ -0,0 +1,632 @@
+    int divisor = Math.max(1, generator.nextInt(original));
+
+    // TODO: #659 We should be able to set max limit of iterations.
+    final int iterations = original / divisor;
+    // A left over number that will be added after for loop is executed. The binary
+    // expression responsible to add a remainder to the variable will be inserted after for
+    // loop statement.
+    final int remainder = original % divisor;
+
+    // Values of numbers which will be used for the for loop.
+    final Value divisorValue = new NumericValue(BasicType.INT, Optional.of(divisor));
+    final Value iterationValue = new NumericValue(BasicType.INT,
+        Optional.of(isIncrement ? 0 : iterations));
+    final Value conditionValue = new NumericValue(BasicType.INT,
+        Optional.of(isIncrement ? iterations : 0));
+
+    final Stmt init = new DeclarationStmt(new VariablesDeclaration(BasicType.INT,
+        new VariableDeclInfo("i", null,
+            new Initializer(generateExpr(factManager, functionDefinition, stmt, iterationValue))))
+    );
+    final Expr condition = new BinaryExpr(new VariableIdentifierExpr("i"),
+        generateExpr(factManager, functionDefinition, stmt, conditionValue),
+        isIncrement ? BinOp.LT : BinOp.GE);
+    final Expr increment = new UnaryExpr(new VariableIdentifierExpr("i"),
+        isIncrement ? UnOp.POST_INC : UnOp.POST_DEC);
+    final Stmt body = new BlockStmt(Arrays.asList(new ExprStmt(new BinaryExpr(
+        new VariableIdentifierExpr(varName),
+        generateExpr(
+            factManager,
+            functionDefinition,
+            stmt,
+            divisorValue),
+        BinOp.ADD_ASSIGN
+    ))), false);
+    final ForStmt forStmt = new ForStmt(init, condition, increment, body);
+    functionDefinition.getBody().insertBefore(stmt, forStmt);
+
+    if (remainder > 0) {
+      final Value remainderValue = new NumericValue((BasicType) value.getType(),
+          Optional.of(remainder));
+      functionDefinition.getBody().insertBefore(stmt, new ExprStmt(new BinaryExpr(
+          new VariableIdentifierExpr(varName),
+          generateExpr(factManager, functionDefinition, stmt, remainderValue),
+          BinOp.ADD_ASSIGN
+      )));
+    }
+    ;
+    final VariableDeclFact variableDeclFact = new VariableDeclFact(variablesDecl,
+        variableDeclInfo, value);
+    factManager.addVariableFact(value, variableDeclFact);
+    return new VariableIdentifierExpr(varName);
+  }
+
+  private Expr generateAdditionValue(FactManager factManager,
+                                     FunctionDefinition functionDefinition,
+                                     Stmt stmt, Value value) {
+    if (!(value instanceof NumericValue)) {
+      return null;
+    }
+
+    if (value.getType() != BasicType.INT && value.getType() != BasicType.FLOAT) {
+      return null;
+    }
+
+    final Pair<Optional<Number>, Optional<Number>> pair = getPairSum(value);
```

Comment:
To allow test case reduction, we will want to eventually wrap all things that generate a value in _GLF_IDENTITY(literal value, ...), e.g. here we should do, if we are representing 42 as 20 + 22:

_GLF_IDENTITY(42, _GLF_IDENTITY(20, expr_generating_20), _GLF_IDENTITY(22, expr_generating_22)))

For now can you open a GitHub issue for this?